File: specs-go/v1/mediatype.go
/*
 *     Copyright 2024 The CNAI Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package v1

const (
	// MediaTypeModelManifest specifies the media type for a model manifest.
	MediaTypeModelManifest = "application/vnd.cnai.model.manifest.v1+json"

	// MediaTypeModelConfig specifies the media type for a model configuration.
	MediaTypeModelConfig = "application/vnd.cnai.model.config.v1+json"

	// MediaTypeModelLayer is the media type used for layers referenced by the manifest.
	MediaTypeModelLayer = "application/vnd.cnai.model.layer.v1.tar"

	// MediaTypeModelLayerGzip is the media type used for gzipped layers
	// referenced by the manifest.
	MediaTypeModelLayerGzip = "application/vnd.cnai.model.layer.v1.tar+gzip"

	// MediaTypeModelDoc specifies the media type for model documentation, including documentation files like `README.md`, `LICENSE`, etc.
	MediaTypeModelDoc = "application/vnd.cnai.model.doc.v1.tar"

	// MediaTypeModelCode specifies the media type for a model code, includes code artifacts like scripts, code files etc.
	MediaTypeModelCode = "application/vnd.cnai.model.code.v1.tar"

	// MediaTypeModelDataset specifies the media type for a model dataset, includes datasets that may be needed for the lifecycle of AI/ML models.
	MediaTypeModelDataset = "application/vnd.cnai.model.dataset.v1.tar"
)

File: specs-python/modelpack/v1/mediatype.py
#     Copyright 2025 The CNCF ModelPack Authors
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""Media type constants matching specs-go/v1/mediatype.go."""

# Artifact type for a model manifest.
ARTIFACT_TYPE_MODEL_MANIFEST = "application/vnd.cncf.model.manifest.v1+json"

# Media type for a model configuration.
MEDIA_TYPE_MODEL_CONFIG = "application/vnd.cncf.model.config.v1+json"

# Model weight media types.
MEDIA_TYPE_MODEL_WEIGHT_RAW = "application/vnd.cncf.model.weight.v1.raw"
MEDIA_TYPE_MODEL_WEIGHT = "application/vnd.cncf.model.weight.v1.tar"
MEDIA_TYPE_MODEL_WEIGHT_GZIP = "application/vnd.cncf.model.weight.v1.tar+gzip"
MEDIA_TYPE_MODEL_WEIGHT_ZSTD = "application/vnd.cncf.model.weight.v1.tar+zstd"

# Model weight config media types.
MEDIA_TYPE_MODEL_WEIGHT_CONFIG_RAW = "application/vnd.cncf.model.weight.config.v1.raw"
MEDIA_TYPE_MODEL_WEIGHT_CONFIG = "application/vnd.cncf.model.weight.config.v1.tar"
MEDIA_TYPE_MODEL_WEIGHT_CONFIG_GZIP = (
    "application/vnd.cncf.model.weight.config.v1.tar+gzip"
)
MEDIA_TYPE_MODEL_WEIGHT_CONFIG_ZSTD = (
    "application/vnd.cncf.model.weight.config.v1.tar+zstd"
)

# Model documentation media types.
MEDIA_TYPE_MODEL_DOC_RAW = "application/vnd.cncf.model.doc.v1.raw"
MEDIA_TYPE_MODEL_DOC = "application/vnd.cncf.model.doc.v1.tar"
MEDIA_TYPE_MODEL_DOC_GZIP = "application/vnd.cncf.model.doc.v1.tar+gzip"
MEDIA_TYPE_MODEL_DOC_ZSTD = "application/vnd.cncf.model.doc.v1.tar+zstd"

# Model code media types.
MEDIA_TYPE_MODEL_CODE_RAW = "application/vnd.cncf.model.code.v1.raw"
MEDIA_TYPE_MODEL_CODE = "application/vnd.cncf.model.code.v1.tar"
MEDIA_TYPE_MODEL_CODE_GZIP = "application/vnd.cncf.model.code.v1.tar+gzip"
MEDIA_TYPE_MODEL_CODE_ZSTD = "application/vnd.cncf.model.code.v1.tar+zstd"

# Model dataset media types.
MEDIA_TYPE_MODEL_DATASET_RAW = "application/vnd.cncf.model.dataset.v1.raw"
MEDIA_TYPE_MODEL_DATASET = "application/vnd.cncf.model.dataset.v1.tar"
MEDIA_TYPE_MODEL_DATASET_GZIP = "application/vnd.cncf.model.dataset.v1.tar+gzip"
MEDIA_TYPE_MODEL_DATASET_ZSTD = "application/vnd.cncf.model.dataset.v1.tar+zstd"
